app/src/main/java/io/mochahub/powermeter/workoutsession/workoutsessiondialog/NewWorkoutDialog.kt

```diff
@@ -0,0 +1,340 @@
+import io.mochahub.powermeter.models.Exercise
+import io.mochahub.powermeter.models.Workout
+import io.mochahub.powermeter.models.WorkoutSession
+import io.mochahub.powermeter.models.WorkoutSet
+import io.mochahub.powermeter.models.addSet
+import io.mochahub.powermeter.models.removeSet
+import io.mochahub.powermeter.models.setReps
+import io.mochahub.powermeter.models.setWeight
+import io.mochahub.powermeter.models.updateExercise
+import kotlinx.android.synthetic.main.dialog_new_workout.addWorkoutBtn
+import kotlinx.android.synthetic.main.dialog_new_workout.newWorkoutDateText
+import kotlinx.android.synthetic.main.dialog_new_workout.newWorkoutNameText
+import kotlinx.android.synthetic.main.dialog_new_workout.newWorkoutToolbar
+import kotlinx.android.synthetic.main.fragment_exercise.recyclerView
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import java.time.Instant
+import java.util.Calendar
+import java.util.Date
+import kotlin.math.absoluteValue
+
+class NewWorkoutDialog : WorkoutController.AdapterCallbacks, DialogFragment() {
+
+    private val args: NewWorkoutDialogArgs by navArgs()
+    private var shouldSave = true
+    private lateinit var workoutController: WorkoutController
+    private lateinit var viewModel: NewWorkoutViewModel
+    private var exercises = listOf<ExerciseEntity>()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setStyle(
+            STYLE_NORMAL,
+            R.style.FullScreenDialog
+        )
+
+        viewModel =
+            NewWorkoutViewModel(
+                db = AppDatabase(requireContext())
+            )
+        CoroutineScope(Dispatchers.IO).launch {
+            exercises = viewModel.getExercises()
+        }.invokeOnCompletion {
+            workoutController =
+                WorkoutController(
+                    requireContext(),
+                    exercises.map { it.name },
+                    this
+                )
+        }
+    }
+
+    override fun onStart() {
+        super.onStart()
+
+        if (dialog != null) {
+            val width = ViewGroup.LayoutParams.MATCH_PARENT
+            val height = ViewGroup.LayoutParams.MATCH_PARENT
+            dialog!!.window!!.setLayout(width, height)
+        }
+    }
+
+    override fun onConfigurationChanged(newConfig: Configuration) {
+        super.onConfigurationChanged(newConfig)
+        workoutController.setData(viewModel.workouts)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.dialog_new_workout, container, false)
+    }
+
+    override fun onDismiss(dialog: DialogInterface) {
+        super.onDismiss(dialog)
+
+        if (shouldSave) {
+            saveWorkoutSession()
+        } else {
+            shouldSave = true
+        }
```

Comment:
Whats the reason for this?

Reply:
It's not needed. Removed it here https://github.com/power-meter/android-client/pull/32/commits/1ed3f0707e96844d84f3a929b0855a51c90ae533